include/osquery/tables.h

```diff
@@ -759,6 +759,33 @@ class TablePlugin : public Plugin {
     return QueryData();
   }
 
+  /// Callback for DELETE statements
+  virtual QueryData delete_(QueryContext& context,
+                            const PluginRequest& request) {
+    static_cast<void>(context);
+    static_cast<void>(request);
+
```

Comment:
Let's replicate the existing error messages: `Error: table osquery_info may not be modified`.

Reply:
With the latest commit, tables can now return custom error messages. By default (i.e. when the insert/update/delete methods are not implemented) read only is returned, showing the message you mentioned